### PR TITLE
Feature: update theme navigation block ui

### PR DESCRIPTION
**Pull request by kaitlinbolling**
Updates the Theme Navigation block to:
* support wide & full alignment
* render a `nav` element without an extra menu container
* render a Placeholder component in the editor
* update styles to be more generic (and let the theme handle layout)

### CODE REVIEW

File: plugin-templates/blocks/theme-navigation/edit.tsx [hunk 1]
```diff
@@ -64,14 +65,22 @@ export default function Edit({
...
 
   // No menu locations have been registered.
   if (menuLocations.length === 0) {
     return (
       <div {...blockProps}>
-        <h3>{ __('Please register menu locations.', 'create-wordpress-plugin') }</h3>
+        <Placeholder
+          icon={ menu }
+          label={ __('Theme Navigation', 'create-wordpress-project') }
+          instructions={ __('Please register menu locations.', 'create-wordpress-project') }
```
Comment (stevenslack) [inline plugin-templates/blocks/theme-navigation/edit.tsx]: The textdomain in the `block.json` is `create-wordpress-plugin` but here it is `create-wordpress-project`.

File: plugin-templates/blocks/theme-navigation/edit.tsx [hunk 2]
```diff
@@ -82,9 +91,11 @@ export default function Edit({
         menuLocation ? (
           <ServerSideRender block="create-wordpress-plugin/theme-navigation" attributes={{ menuLocation }} />
         ) : (
-          <h3>
-            {__('Please select a menu location.', 'create-wordpress-plugin')}
-          </h3>
+          <Placeholder
+            icon={ menu }
+            label={ __('Theme Navigation', 'create-wordpress-project') }
+            instructions={ __('Please select a menu location.', 'create-wordpress-project') }
```
Comment (stevenslack) [inline plugin-templates/blocks/theme-navigation/edit.tsx]: Same issue here for the textdomain

**Review by stevenslack — COMMENTED** [review]
The text domain value in the `edit.tsx` file should be `create-wordpress-plugin`

**Review by efuller — APPROVED** [review]
⚡

**Review by mogmarsh — APPROVED** [review]
👍

**Pull request merged**